components/remote-environment-broker/internal/config/flavor.go

```diff
@@ -0,0 +1,37 @@
+package config
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// BrokerFlavor aggregates information about mode REB is running: if it is cluster scoped or namespaced scoped
+type BrokerFlavor struct {
+	clusterScoped      bool
+	clusterBrokerName  string
+	nsBrokerURLPattern *regexp.Regexp
+}
+
+// NewBrokerFlavorFromConfig creates BrokerFlavor from configuration
+func NewBrokerFlavorFromConfig(cfg *Config) *BrokerFlavor {
+	return &BrokerFlavor{
+		clusterBrokerName:  cfg.ClusterScopedBrokerName,
+		clusterScoped:      cfg.ClusterScopedBroker,
+		nsBrokerURLPattern: regexp.MustCompile("reb-ns-for-([a-z][a-z0-9-]*)\\."),
+	}
+}
+
+// GetNsFromBrokerURL extracts namespace from broker URL
+func (bf *BrokerFlavor) GetNsFromBrokerURL(url string) (string, error) {
+	out := bf.nsBrokerURLPattern.FindStringSubmatch(url)
+	if len(out) == 0 {
+		return "", fmt.Errorf("url:%s does not match pattern %s", url, bf.nsBrokerURLPattern.String())
+	}
+	// out[0] = matched regexp, out[1] = matched group in bracket
+	return out[1], nil
```

Comment:
It will never happen that we match whole expression and don't match subexpression, because regexp is defined in that way that group match is required [a-z].
But to not confuse others I change if